### ISSUE TITLE
Humbug reports few bad tests

### ISSUE THREAD

**Issue by Ocramius** [psr7-sessions/storageless#49]
Here's a simple `humbuglog.txt`:

```
Humbug running test suite to generate logs and code coverage data...
Humbug has completed the initial test run successfully.
Tests: 192 Line Coverage: 89.19%
Humbug is analysing source files...
Mutation Testing is commencing on 4 files...
(.: killed, M: escaped, S: uncovered, E: fatal error, T: timed out)
SSSSSS...........M.M.M..........
32 mutations were generated:
      23 mutants were killed
       6 mutants were not covered by tests
       3 covered mutants were not detected
       0 fatal errors were encountered
       0 time outs were encountered

Metrics:
    Mutation Score Indicator (MSI): 72%
    Mutation Code Coverage: 81%
    Covered Code MSI: 88%

Remember that some mutants will inevitably be harmless (i.e. false positives).
Time: 5.83 seconds Memory: 6.00MB
Humbug results are being logged as JSON to: humbuglog.json
Humbug results are being logged as TEXT to: humbuglog.txt

------
Uncovered
------


1) Mutator \Humbug\Mutator\ReturnValue\NewObject on \PSR7Session\Http\SessionMiddleware::fromSymmetricKeyDefaults() in /Users/ocramius/Documents/Projects/PSR7Session/src/PSR7Session/Http/SessionMiddleware.php on line 118
2) Mutator \Humbug\Mutator\Boolean\TrueValue on \PSR7Session\Http\SessionMiddleware::fromSymmetricKeyDefaults() in /Users/ocramius/Documents/Projects/PSR7Session/src/PSR7Session/Http/SessionMiddleware.php on line 123
3) Mutator \Humbug\Mutator\Boolean\TrueValue on \PSR7Session\Http\SessionMiddleware::fromSymmetricKeyDefaults() in /Users/ocramius/Documents/Projects/PSR7Session/src/PSR7Session/Http/SessionMiddleware.php on line 124
4) Mutator \Humbug\Mutator\ReturnValue\NewObject on \PSR7Session\Http\SessionMiddleware::fromAsymmetricKeyDefaults() in /Users/ocramius/Documents/Projects/PSR7Session/src/PSR7Session/Http/SessionMiddleware.php on line 145
5) Mutator \Humbug\Mutator\Boolean\TrueValue on \PSR7Session\Http\SessionMiddleware::fromAsymmetricKeyDefaults() in /Users/ocramius/Documents/Projects/PSR7Session/src/PSR7Session/Http/SessionMiddleware.php on line 150
6) Mutator \Humbug\Mutator\Boolean\TrueValue on \PSR7Session\Http\SessionMiddleware::fromAsymmetricKeyDefaults() in /Users/ocramius/Documents/Projects/PSR7Session/src/PSR7Session/Http/SessionMiddleware.php on line 151


------
Escapes
------


1) \Humbug\Mutator\Boolean\FalseValue
Diff on \PSR7Session\Http\SessionMiddleware::shouldTokenBeRefreshed() in /Users/ocramius/Documents/Projects/PSR7Session/src/PSR7Session/Http/SessionMiddleware.php:
--- Original
+++ New
@@ @@
         if (null === $token) {
-            return false;
+            return true;
         }

         if (! $token->hasClaim(self::ISSUED_AT_CLAIM)) {
             return false;
         }



2) \Humbug\Mutator\Boolean\FalseValue
Diff on \PSR7Session\Http\SessionMiddleware::shouldTokenBeRefreshed() in /Users/ocramius/Documents/Projects/PSR7Session/src/PSR7Session/Http/SessionMiddleware.php:
--- Original
+++ New
@@ @@
         if (! $token->hasClaim(self::ISSUED_AT_CLAIM)) {
-            return false;
+            return true;
         }

         return time() >= ($token->getClaim(self::ISSUED_AT_CLAIM) + $this->refreshTime);
     }




3) \Humbug\Mutator\ConditionalBoundary\GreaterThanOrEqualTo
Diff on \PSR7Session\Http\SessionMiddleware::shouldTokenBeRefreshed() in /Users/ocramius/Documents/Projects/PSR7Session/src/PSR7Session/Http/SessionMiddleware.php:
--- Original
+++ New
@@ @@

-        return time() >= ($token->getClaim(self::ISSUED_AT_CLAIM) + $this->refreshTime);
+        return time() > ($token->getClaim(self::ISSUED_AT_CLAIM) + $this->refreshTime);
     }


     /**
      * @param SessionInterface $sessionContainer
      *


```

This basically means that we either didn't cover the code completely, or that we didn't assert enough on the outcomes.

**Comment by lcobucci** [comment]
This was solved on #55, right @Ocramius ?

**Comment by lcobucci** [comment]
Closing since [humbug log is puurrfect now](https://travis-ci.org/psr7-sessions/storageless/jobs/222057292#L263)
